rts/motoko-rts/src/gc/incremental/phases/mark_increment.rs

```diff
@@ -49,6 +57,8 @@ impl<'a, M: Memory + 'a> MarkIncrement<'a, M> {
     pub unsafe fn mark_completed(state: &State) -> bool {
         let mark_state = state.mark_state.as_ref().unwrap();
         debug_assert!(!mark_state.complete || mark_state.mark_stack.is_empty());
+        #[cfg(feature = "enhanced_orthogonal_persistence")]
+        debug_assert!(mark_state.weak_ref_registry.is_empty());
```

Comment:
I think it should be:
```suggestion
        debug_assert!(!mark_state.complete || mark_state.weak_ref_registry.is_empty());
```
Weak registry can be non-empty during mark phase.